### ISSUE TITLE
Security update in 4.0.2 broke saving of multi-value form elements

### ISSUE THREAD

**Issue by dmitryd** [frappant/frp_form_answers#57]
Multi-value form elements are arrays. Therefore `htmlspecialchars()` [here](https://github.com/frappant/frp_form_answers/blob/master/Classes/Domain/Finishers/SaveFormToDatabaseFinisher.php#L132) returns `null` and the value is lost.

<img width="537" alt="frp" src="https://user-images.githubusercontent.com/306133/220632052-10db9788-965a-4d25-826e-d306d97e31a3.png">

This code does not look correct. Values should only be html-escaped when they are displayed, not when they are stored to the database.

This code also breaks html output in the module. If you type `Test<script>alert('Hello')</script>` into a form field, you should see it exactly like this in the module. Instead what you see in the module is double-escaped output: 

| Form | Module |
|--|--|
| <img width="187" alt="1" src="https://user-images.githubusercontent.com/306133/220632697-caee6794-75f7-4c14-9eff-25739113be1e.png"> | <img width="288" alt="2" src="https://user-images.githubusercontent.com/306133/220632836-03ddd294-36d3-440a-af5f-e2edaa0fb69c.png"> |

Fluid automatically applies `htmlspecialchars` to all variables. Therefore the code [here](https://github.com/frappant/frp_form_answers/blob/master/Resources/Private/Backend/Partials/FormEntry/Properties.html#L39) and in other places of this file also does not look correct because the value is already escaped by fluid. Same about another html template. So the change in html files is not needed.

I believe the only needed change is in `Classes/Form/FormAnswersJsonElement.php` and all others are not needed for the above mentioned reasons. However `FormAnswersJsonElement` is also broken for the multivalue form fields because `htmlspecialchars` does not work on array (bad code is `implode(",", htmlspecialchars($fieldValue['value']))` because `$fieldValue['value']` is array!).

If I completely remove all `htmlspecialchars` from the finisher and html template, I see correct results in the finisher:

<img width="267" alt="3" src="https://user-images.githubusercontent.com/306133/220635603-db092796-54c0-433c-baf3-522bac1483aa.png">

To summarize:
- fix `Classes/Form/FormAnswersJsonElement.php`
- revert all other changes
